### ISSUE TITLE
500er Fehler beim API-Call im Backend

### ISSUE THREAD

**Issue by tyrant88** [alexplusde/link_checker#12]
![Zwischenablage-1](https://user-images.githubusercontent.com/1277494/210778204-56587db5-77c3-43d3-9c1b-409326874b28.jpg)

Du solltest die response vorher checken mit `if ($response->isOk()) {`

Generell empfehle ich die "saubere" PHP Funktion parse_url() zu nutzen, um die URL zu zerlegen (rex_socket selbst nutzt das auch)

https://www.php.net/manual/en/function.parse-url.php

**Comment by tyrant88** [comment]
Nee, es hilft doch nur ein `if($response->isOk() && '' != $response->getBody()){`

**Comment by alxndr-w** [comment]
Das ist ja hervorragend, genau so etwas habe ich in dem Moment gesucht.

**Comment by alxndr-w** [comment]
@tyrant88 magst du einen PR dafür vorbereiten?

https://github.com/alexplusde/link_checker/blob/344342268c108295abab8171e3939ca029f2e617/lib/link_checker_source.php#L45-L70

sowie 

https://github.com/alexplusde/link_checker/blob/344342268c108295abab8171e3939ca029f2e617/lib/link_checker_source.php#L40

**Comment by tyrant88** [comment]
Ich hab/hatte gerade keine Zeit, deshalb nur issue. Ich kann auch nicht beurteilen, WARUM der body da leer ist.

Liest das AddOn eigentlich nur bei installation die Sitemap ein? Wie findet es denn neuangelegte Artikel?

**Comment by alxndr-w** [comment]
Siehe #5
